Manubot sections

dhimmel opened the pull request:
I'm off for the weekend!

dhimmel committed:
Manubot markdown & citation paragraphs

agitter commented:
I'll review this weekend

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,23 @@ Although we requested that some authors participate for their specific expertise
 `TODO: confirm "most"`
 To coordinate this effort, we developed a manuscript writing process using the Markdown language, the GitHub software development platform [@url:https://github.com/greenelab/deep-review/], and our new Manubot tool [@url:https://github.com/greenelab/manubot-rootstock; @url:https://github.com/greenelab/manubot] for automating manuscript generation.
 
+### Manubot
```
Should we make our sub-section titles more descriptive?  Maybe `Writing manuscripts with Manubot`?

dhimmel replied:
Yes. Once I complete the writing, we should think about restructuring paragraphs and headings.

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,23 @@ Although we requested that some authors participate for their specific expertise
-### Collaborative writing and reviewing
+We developed Manubot, a system for writing scholarly manuscripts via GitHub.
+With Manubot, manuscripts are written as plain-text markdown files, which is well suited for version control using git.
+The markdown standard itself provides limited yet crucial formatting syntax, including the ability to embed images and format text via bold, italics, hyperlinks, headers, blockquotes, inline code, codeblocks, blockquotes, and numbered or bulleted lists.
```
`blockquotes` is repeated.  We could skip some of the formatting options that are less relevant for manuscripts like `inline code` and `codeblocks`.

dhimmel replied:
> We could skip some of the formatting options that are less relevant for manuscripts like inline code and codeblocks.

I think all of these formatting options are relevant and important to scholarly writing. They are just traditionally absent from manuscripts because Word does a poor job supporting them.

agitter replied:
Point taken

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,23 @@ Although we requested that some authors participate for their specific expertise
+In addition, Manubot relies on Pandoc's extended markdown to enable citations, tables, captions, and equations specified using the popular TeX math syntax.
+
+Manubot includes an additional layer of citation processing, currently unique to the system.
+All citations point to standard identifier, for which Manubot automatically retrieves bibliographic metadata.
```
`to a standard`

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,23 @@ Although we requested that some authors participate for their specific expertise
+Currently, citations to DOIs (Digital Object Idenfiers), PubMed identifiers, arXiv identifiers, and URLs (web addresses) are supported.
+Metadata is retrieved using DOI [Content Negotiation](https://citation.crosscite.org/docs.html), NCBI's [Citation Exporter](https://www.ncbi.nlm.nih.gov/pmc/tools/ctxp/), the [arXiv API](https://arxiv.org/help/api/index), or [Greycite](http://greycite.knowledgeblog.org/) [@arxiv:1304.7151v1].
+Metadata is exported to [CSL JSON Items](http://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html#items), which enables specifying the bibliographic format using [Citation Styles Language](http://citationstyles.org/) (CSL).
+As a result, users can choose from thousands of existing CSL styles or use Manubot's default style.
```
We don't have much space, but should we briefly explain what CSL controls for someone who is not familiar?  We could provide an example of determining how many authors to show, how to format names, etc.

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,23 @@ Although we requested that some authors participate for their specific expertise
+
+Manubot includes an additional layer of citation processing, currently unique to the system.
+All citations point to standard identifier, for which Manubot automatically retrieves bibliographic metadata.
+Currently, citations to DOIs (Digital Object Idenfiers), PubMed identifiers, arXiv identifiers, and URLs (web addresses) are supported.
```
The next few sentences feel heavy on jargon.  I don't have a good idea for avoiding it though.

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,23 @@ Although we requested that some authors participate for their specific expertise
+The markdown standard itself provides limited yet crucial formatting syntax, including the ability to embed images and format text via bold, italics, hyperlinks, headers, blockquotes, inline code, codeblocks, blockquotes, and numbered or bulleted lists.
+In addition, Manubot relies on Pandoc's extended markdown to enable citations, tables, captions, and equations specified using the popular TeX math syntax.
+
+Manubot includes an additional layer of citation processing, currently unique to the system.
```
The citation processing is a huge feature.  Should we emphasize it more?  Maybe include a figure where we show how a DOI reference expands to the citation automatically?  Or include an example citation in the text here?

dhimmel replied:
Yes we should demonstrate the capability. I will think of whether a figure or textual example makes the most sense. Currently, manubot replaces citations in code-formatted text (which is wrong, but hard to fix).

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,23 @@ Although we requested that some authors participate for their specific expertise
+Manubot includes an additional layer of citation processing, currently unique to the system.
+All citations point to standard identifier, for which Manubot automatically retrieves bibliographic metadata.
+Currently, citations to DOIs (Digital Object Idenfiers), PubMed identifiers, arXiv identifiers, and URLs (web addresses) are supported.
+Metadata is retrieved using DOI [Content Negotiation](https://citation.crosscite.org/docs.html), NCBI's [Citation Exporter](https://www.ncbi.nlm.nih.gov/pmc/tools/ctxp/), the [arXiv API](https://arxiv.org/help/api/index), or [Greycite](http://greycite.knowledgeblog.org/) [@arxiv:1304.7151v1].
```
I had been using URL citations instead of links.  We should be consistent.  Links might be best.

dhimmel replied:
I think it makes most sense to hyperlink to services or entities. Citations make sense for "citeable work"... things that generally have at least 3 of the following 4 pieces of metadata: authors, title, publisher, or date.

If a service also has a corresponding citable work, such as Greycite, then you can link and cite.

Along these lines, I'm in favor of just hyperlinking to GitHub repos. When referring to Manubot manuscripts (hosted via GH Pages) a citation is appropriate.

@agitter do you agree?

agitter replied:
@dhimmel Yes, I agree with that.  I created #11 to remind me to convert some of my existing URL citations.

agitter reviewed (changes requested):
I like this.  It's compact but does a good job explaining the approach, which is just what we need.  Most of my comments are discussion points, not strong recommendations.

I'd like to add a sentence or two about how Manubot + continuous analysis are used to check references, build the manuscript, and share HTML and PDF versions.  That was important for deep review.

I also wonder whether this sub-section might fit better after `Contribution workflow`.

Should we discuss other work related to Manubot in the `Discussion`?  That's where I currently have a placeholder.

dhimmel committed:
Revise Manubot citation / reference section

Addresses comment by @agitter:

> should we briefly explain what CSL controls for someone who is
not familiar

dhimmel committed:
Export format paragraph

dhimmel committed:
Continous integration, deployment, timestamping

agitter commented:
@dhimmel is this ready for re-review?

dhimmel commented:
> @dhimmel is this ready for re-review?

Yes. I do plan on adding a few more paragraphs, but you should still be able to review the current state.

agitter commented:
> Yes. I do plan on adding a few more paragraphs, but you should still be able to review the current state.

We don't have much more space in this format.  The [guidelines](https://github.com/greenelab/meta-review/blob/master/GUIDELINES.md) say

> There are no strict limits on the length, but authors are advised to stay below 2000 words

The word count of 2390 in the build is not accurate, but it suggests we're nearly maxed out.  Trimming TODOs will help with a better estimate.

dhimmel commented:
> There are no strict limits on the length, but authors are advised to stay below 2000 words, two display items (figures, tables etc) and 20 references if possible.

In general, I think eLife is very reasonable. So if we think that the content would be best communicated in over 2000 words, I think we should do that. My inclination would be to write the article that we want to write. And if eLife judges it to be unnecessarily verbose, we can cut it down.

agitter commented:
Okay, I won't worry about length in this first draft.

We should be careful about making it broadly accessible.  All of us are familiar with the jargon used here so we may want to have someone with a different background read the complete draft.

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+We developed Manubot, a system for writing scholarly manuscripts via GitHub.
+With Manubot, manuscripts are written as plain-text markdown files, which is well suited for version control using git.
+The markdown standard itself provides limited yet crucial formatting syntax, including the ability to embed images and format text via bold, italics, hyperlinks, headers, inline code, codeblocks, blockquotes, and numbered or bulleted lists.
+In addition, Manubot relies on Pandoc's extended markdown to enable citations, tables, captions, and equations specified using the popular TeX math syntax.
```
Introduce Pandoc?  Something like `Manubot relies on Pandoc, software for converting document formats, to extend markdown and enable`

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+As a result, users can choose from thousands of existing CSL styles or use Manubot's default style.
+Styles define how references are constructed from bibliographic metadata, controlling layout details such as the max number of authors to list per reference.
+Thousands of journals have [predefined styles](http://editor.citationstyles.org/searchByName/).
+As a result, adopting the specific bibliographic format required by a journal usually just requires specifying the style's source URL in the Manubot's configuration.
```
` Manubot's` or `the Manubot` sounds better to me

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+Thousands of journals have [predefined styles](http://editor.citationstyles.org/searchByName/).
+As a result, adopting the specific bibliographic format required by a journal usually just requires specifying the style's source URL in the Manubot's configuration.
+
+Manubot uses [Pandoc](https://pandoc.org/) to convert manuscripts from markdown source to HTML (and optionally DOCX) output.
```
Move link to the first mention of Pandoc.

Can this be simplified to `to HTML, PDF, and optionally DOCX output`?

dhimmel replied:
Will consider. Technically PDF is converted from the HTML using WeasyPrint. But that may be an irrelevant detail.

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+For example, Manubot intentionally does not rely on LaTeX, although users are free to export manuscripts to LaTeX themselves.
+In a future release, Pandoc [will support](https://github.com/greenelab/manubot-rootstock/pull/51#issuecomment-321949622) export to [Journal Article Tag Suite](https://jats.nlm.nih.gov/) (JATS).
+JATS is a standard XML format for scholarly articles that is used by publishers, archives, and text miners [@url:http://www.niso.org/standards/z39-96-2015/; @doi:10.6087/kcse.2014.1.99; @doi:10.1080/00987913.2012.10765464].
+For now however, the primary Manubot output is HTML intended to be viewed in a web browser.
```
This line and the next could be removed if you simplify the first sentence in the paragraph.

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+As a result, adopting the specific bibliographic format required by a journal usually just requires specifying the style's source URL in the Manubot's configuration.
+
+Manubot uses [Pandoc](https://pandoc.org/) to convert manuscripts from markdown source to HTML (and optionally DOCX) output.
+Pandoc supports conversion between a large number of formats, offering Manubot users broad interoperability and preventing platform lock-in.
```
Should we define `platform`?

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+
+Manubot performs continuous publication: every update to a manuscript's source is automatically reflected in the online outputs.
+The approach uses continuous integration (CI) [@doi:10.1038/550143a; @doi:10.1038/nbt.3780], specifically [Travis CI](https://travis-ci.org/) at the moment, to monitor for changes.
+When changes occur, the CI service rebuilds the manuscript.
```
Is `generates updated HTML and PDF documents` more understandable than `rebuilds the manuscript`?

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+
+For this article, the source GitHub repository is https://github.com/greenelab/meta-review.
+When this repository changes, Travis CI [rebuilds](https://travis-ci.org/greenelab/meta-review) the manuscript.
+If successful, the output is deployed back to GitHub (to dedicated [`output`](https://github.com/greenelab/meta-review/tree/output) and [`gh-pages`](https://github.com/greenelab/meta-review/tree/gh-pages) branches).
```
The branches parenthetical is getting fairly technical

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+If successful, the output is deployed back to GitHub (to dedicated [`output`](https://github.com/greenelab/meta-review/tree/output) and [`gh-pages`](https://github.com/greenelab/meta-review/tree/gh-pages) branches).
+As a result, https://greenelab.github.io/meta-review stays up to date with the latest HTML manuscript.
+
+Manubot uses [OpenTimestamps](https://opentimestamps.org/) to timestamps the HTML and PDF outputs on the Bitcoin blockchain before deploying to GitHub.
```
`timestamp`

agitter commented on content/02.main-text.md:
```diff
@@ -20,8 +20,54 @@ Although we requested that some authors participate for their specific expertise
+Manubot uses [OpenTimestamps](https://opentimestamps.org/) to timestamps the HTML and PDF outputs on the Bitcoin blockchain before deploying to GitHub.
+This procedure allows one to retrospectively prove that a manuscript version existed prior to its blockchain-verifiable timestamp [@arxiv:1502.04015v1; @url:https://www.bgcarlisle.com/blog/2014/08/25/proof-of-prespecified-endpoints-in-medical-research-with-the-bitcoin-blockchain/; @url:http://blog.dhimmel.com/irreproducible-timestamps/; @url:http://git.dhimmel.com/bitcoin-whitepaper/].
+The implications for scientific writing are twofold.
+First, scientific precedence can now be indisputably established.
```
Is that necessarily a good thing?  An [eLife article](https://doi.org/10.7554/eLife.28699) recently pushed back against scientific precedence.

dhimmel replied:
The implications are probably most important legally, when precedence can effect the outcome of an intellectual property dispute or some other question. Personally when evaluating science, I care for quality over novelty. For example, I may cite a more thorough and accessible study over an earlier study. In short, being able to establish precedence is potentially valuable for authors. We're not taking a position of the use of precedence or novelty when judging scientific output. The extent novelty should be used for evaluation (or in legal cases) is a separate question that we don't need to answer.

agitter reviewed (commented):
More intermediate comments.  I'll give a final review when you are done making your planned changes.

dhimmel committed:
Revisions based on @agitter review

https://github.com/greenelab/meta-review/pull/9#pullrequestreview-67221216

dhimmel committed:
Add manubot implications paragraph

dhimmel commented:
Okay this section now contains the main points I want to make about the Manubot. We all probably agree that the various manuscripts sections could be better integrated. Perhaps the right approach would be to merge this and #12 once the in-section issues are fixed. Then we can do more transpositions and section labeling.

agitter commented:
Looks good to me.  After we merge #12 and reorganize, I'd like to have a third party review the draft.  I'm wondering whether parts of the Manubot section will be understandable to an undergrad.
